types/camunda-external-task-client-js/index.d.ts

```diff
@@ -0,0 +1,149 @@
+        asyncResponseTimeout?: number;
+        interceptors?: Interceptor | Interceptor[];
+        use?: Middleware | Middleware[];
+    });
+
+    start(): void;
+
+    stop(): void;
+
+    subscribe(
+        topic: string,
+        options: SubscribeOptions,
+        handler: Handler,
+    ): TopicSubscription;
+
+    subscribe(
+        topic: string,
+        handler: Handler,
+    ): TopicSubscription;
+}
+
+export class Variables {
+    get(variableName: string): Value;
+
+    getTyped(variableName: string): TypedValue;
+
+    getAll(): Value[];
+
+    getAllTyped(): TypedValue[];
+
+    set(variableName: string, value: Value): Variables;
+
+    setTyped(variableName: string, typedValue: TypedValue): Variables;
+
+    setAll(values: { [key: string]: Value }): Variables;
+
+    setAllTyped(typedValues: { [key: string]: TypedValue }): Variables;
+}
+
+type Handler = (args: HandlerArgs) => void;
+
+interface HandlerArgs {
+    task: Task;
+    taskService: TaskService;
+}
+
+interface Task {
+    variables: Variables;
+
+    // These are not guaranteed by package documentation, but are returned according to REST API docs
+    activityId?: string;
+    activityInstanceId?: string;
+    businessKey?: string;
+    errorDetails?: string;
+    errorMessage?: string;
+    executionId?: string;
+    id?: string;
+    lockExpirationTime?: string;
+    priority?: number;
+    processDefinitionId?: string;
+    processDefinitionKey?: string;
+    processInstanceId?: string;
+    retries?: number;
+    tenantId?: string;
+    topicName?: string;
+    workerId?: string;
+}
+
+type Value = any;
+
+interface TypedValue {
+    value: Value;
+    type: string;
+    valueInfo: object;
+}
+
+interface TaskService {
+    complete(
+        task: Task,
+        processVariables?: Variables,
+        localVariables?: Variables,
+    ): void;
+
+    handleFailure(task: Task, options?: HandleFailureOptions): void;
+
+    handleBpmnError(
+        task: Task,
+        errorCode: string,
+        errorMessage?: string,
+        variables?: Variables,
+    ): void;
+
+    extendLock(task: Task, newDuration: number): void;
+
+    unlock(task: Task): void;
+}
+
+interface HandleFailureOptions {
+    errorMessage?: string;
+    errorDetails?: string;
+    retires?: number;
+    retryTimeout?: number;
+}
+
+interface SubscribeOptions {
+    lockDuration?: number;
+    variables?: any[];
+    businessKey?: string;
+    processDefinitionId?: string;
+    processDefinitionIdIn?: string;
+    processDefinitionKey?: string;
+    processDefinitionKeyIn?: string;
+    processDefinitionVersionTag?: string;
+    withoutTenantId?: boolean;
+    tenantIdIn?: string[];
+}
+
+interface TopicSubscription {
+    unsubscribe: () => void;
+}
+
+type Interceptor = <T = any>(config: T) => T;
+
+type Middleware = (client: Client) => void;
+
+type Logger = Middleware & {
+    success(text: string): void;
+
+    error(text: string): void;
+};
+
+export const logger: Logger;
+
+export {};
```

Comment:
The module kind is slightly wrong here. You’re representing named ES2015 exports, but the target package is a CommonJS module exporting an object. What you want to do is declare a namespace, put your exports in there, and export the namespace as a CommonJS module:

```ts
declare namespace CamundaExternalTaskClient { // or whatever, name is flexible here
  export class Client {
    // ...
  }
  export {}; // Although really it’s nice to export your support types
}

export = CamundaExternalTaskClient;

Reply:
I've applied your remarks, thanks for the review!
Unfortunately, I needed to silence the `export-just-namespace` rule, and I feel a little bad about this. Is there a better way how this could be handled?

Reply:
Oh no, I think I told you the wrong thing! I forgot TypeScript actually treats these equivalently as long as there’s no default export. Modules are hard 🙈. You had it right the first time, I’m sorry!

Reply:
No problem :smile: I've moved it back to top-level, and also exported all support types as you suggested.
Let me know if there is something more I can do.